src/trackers/helpers/cookies.js

```diff
@@ -0,0 +1,47 @@
+
+/**
+ * Returns true iff this savedCall is a valid call to document.cookie
+ * @param {Object} savedCall data
+ */
+function isSavedCookieSetterCall({description, arguments: args}) {
+    return description === 'Document.cookie setter' &&
+                typeof args !== 'boolean' &&
+                args.length > 0 &&
+                typeof args[0] === 'string'
+}
+
+/**
+ * Parse a cookie string as passed to document.cookie setter and extract its attributes
+ * @param {string} cookieString 
+ */
+function parseCookie(cookieString) {
+    const parsed = cookieString.split(';').map(pair => pair.split('=', 2).map(s => s.trim()))
+    return parsed.reduce((info, [k, v], i) => {
+        if (i === 0) {
+            info.name = k
+            info.value = v
+        } else if (k) {
+            info[k.toLowerCase()] = v
+        }
+        return info
+    }, {})
+}
+
+/**
+ * Calculate the cookie time to live, given the timestamp when the cookie was set.
+ * @param {Object} cookie - parsed cookie
+ * @param {Number} setAtTs - timestamp in ms when the cookie was set
+ * @returns TTL of this cookie in seconds from setAtTs.
+ */
+function calculateCookieTtl(cookie, setAtTs) {
```

Comment:
what will happen with a session cookie here?

Reply:
Good point. I added a test to cover this - session cookies will have a ttl of `NaN`, which I think actually makes sense. This will differentiate session cookies from any cookie with an expiry.

The interesting feature of using `NaN`, is that when we calculate the median at the end, if over half of the cookies are session cookies, we'll pick `NaN` as the ttl, showing that these are session cookies in the data.

Reply:
One edge case is when there are two middle values and one of them is NaN and the other one is a proper number. But yeah, I think that NaN in the general case is better than using 0 or infinity here.